chapter10/ajd/exercise1.pl

```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Data::Dumper;
+
+# Rewrite this so it doesn't annoy the monkey:
+chdir;
+my @sorted = sort { -s $a <=> -s $b } glob '*';
+print Dumper( \@sorted );
+
+my @schwartz = map { $_->{name} }
+             sort { $a->{size} <=> $b->{size} }
+             map { { name => $_, size => -s $_ } }
+             glob '*';
```

Comment:
The hash solution is much more readable. I'd be curious to see how the performance differs from the list of arrayref solution.